### PR TITLE
immutably.array.move implementation

### PR DESCRIPTION
**Pull request by creativedeveloper-net**

### CODE REVIEW

File: src/immutably-array/move.test.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        move(input, null, 0, 1);
+    });
+
+    testCase.throws(() => {
+        const input = {};
+        move(input, null, 0, 1);
+    });
+
+    testCase.end();
+});
```
Comment (ennovum) [inline src/immutably-array/move.test.js:84]: I think we should also have tests for scenario where we're trying to move a value from or to a not existing area of the array. E. g.
```
immutably.array.move(['foo', 'bar', 'baz'], null, 1, 3);
immutably.array.move(['foo', 'bar', 'baz'], null, 4, 0);
```

File: src/immutably-array/move.js [hunk 2]
```diff
@@ -4,10 +4,16 @@ function move(input, path, fromIndex, toIndex) {
...
 
+        if (input[fromIndex] === undefined) {
+            throw "immutably-array.move(): trying to move a value from a not existing area of the array.";
+        } else if (input[toIndex] === undefined) {
+            throw "immutably-array.move(): trying to move a value to a not existing area of the array.";
+        }
+
         const output = immutably.clone(input);
 
         output.splice(toIndex, 0, output.splice(fromIndex, 1)[0]);
```
Comment (ennovum) [inline src/immutably-array/move.js]: One of ideas behind `immutably` and `immutably-array` is to rely on native behaviors as much as possible. It includes native errors/exceptions. There is no `move` method in `Array.prototype`, so it's hard to follow it here directly, but indirectly I'm leaning towards a more error-forgiving approach.

E. g. please see what happens when you run `[0, 1, 2].splice(3, 1)`. It returns an empty array and the subject array is untouched. No error. One may argue if this is the right approach, but it is by the spec.

So here I'm thinking that those `throw`s should be removed and this line here should look like this:
`output.splice(toIndex, 0, ...output.splice(fromIndex, 1));`.
This way if the inner splice returns an empty array, the outer splice would do nothing and we'd get the same array as a result. Also if we try to move an existing value to not existing index, the array would expand, as it should I think.

Thoughts?

**Comment by ennovum** [comment]
I decided to merge and carry on on my own. I hope it's OK.

**Comment by creativedeveloper-net** [comment]
ok fair enough
